modules/porous_flow/include/materials/PorousFlowDensityConstBulk.h

```diff
@@ -0,0 +1,71 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef POROUSFLOWDENSITYCONSTBULK_H
+#define POROUSFLOWDENSITYCONSTBULK_H
+
+#include "PorousFlowFluidPropertiesBase.h"
+
+class PorousFlowDensityConstBulk;
+
+template<>
+InputParameters validParams<PorousFlowDensityConstBulk>();
+
+/**
+ * Material designed to calculate fluid density
+ * from porepressure, assuming constant bulk modulus
+ * for the fluid.
+ */
+class PorousFlowDensityConstBulk : public PorousFlowFluidPropertiesBase
+{
+public:
+  PorousFlowDensityConstBulk(const InputParameters & parameters);
+
+protected:
+  virtual void initQpStatefulProperties();
+  virtual void computeQpProperties();
+
+  /**
+   * Density calucaled assuming a constant bulk modulus
+   *
+   * @param pressure gas pressure (Pa)
+   * @return density (kg/m^3)
+   */
+  Real density(Real pressure) const;
+
+  /**
+   * Derivative of the density of a constant bulk density a function of
+   * pressure.
+   *
+   * @param pressure gas pressure (Pa)
+   * @return derivative of density (kg/m^3) with respect to pressure
+   */
+  Real dDensity_dP(Real pressure) const;
```

Comment:
For the record, I'm not a fan of these method names with underscores. But I don't want to be the `PorousFlowMethodNameDictator` :-)